libafl/src/mutators/mutations.rs

```diff
@@ -57,6 +54,23 @@ pub fn buffer_set<T: Clone>(data: &mut [T], from: usize, len: usize, val: T) {
     }
 }
 
+/// Generate a range of values where (upon repeated calls) each index is likely to appear in the
+/// provided range as likely as any other value
+///
+/// The solution for this is to specify a window length, then pretend we can start at indices that
+/// would lead to invalid ranges. Then, clamp the values.
+///
+/// This problem corresponds to: https://oeis.org/A059036
+#[inline]
+pub fn rand_range<S: HasRand>(state: &mut S, upper: usize, max_len: usize) -> Range<usize> {
+    let len = state.rand_mut().next() as usize % max_len + 1;
```

Comment:
It shouldn't be... the addition applies after the modulo, so if max_len is usize::MAX it will not overflow (since the result will be `[0, usize::MAX)`).

This function will only ever be used for ranges, so it makes sense to return a `Range<usize>` since we'll likely only use this to select a range of values.

Reply:
Ah lol yes.. Code is hard